File: lint-arwaky/src-python/contract/diff_result_aggregate.py
from abc import ABC
from pydantic import BaseModel, ConfigDict
from ..taxonomy import Count, FilePathList, RenamedFileList

class GitDiffResultAggregate(BaseModel, ABC):
    """AGGREGATE: Contract for Git diff results."""
    model_config = ConfigDict(arbitrary_types_allowed=True)
    added: FilePathList
    modified: FilePathList
    deleted: FilePathList
    renamed: RenamedFileList
    lintable_files: FilePathList
    all_files: FilePathList
    total_changed: Count

File: lint-arwaky/src-rust/contract/diff_result_aggregate.rs
use super::*;

#[derive(Debug, Clone, Default)]
pub struct GitDiffResultAggregate {
    pub added: FilePathList,
    pub modified: FilePathList,
    pub deleted: FilePathList,
    pub renamed: Vec<RenamedFile>,
    pub lintable_files: FilePathList,
    pub all_files: FilePathList,
    pub total_changed: Count,
}
